### DISCUSSION TITLE
Getting Startify to work with nvim-web-devicons

### DISCUSSION THREAD

**Discussion by Rizhiy** [kyazdani42/nvim-web-devicons#175] [Q&A] [answered]
Hi, `vim-devicons` has function `WebDevIconsGetFileTypeSymbol`, which allows startify to use devicons using:
```vim
function! StartifyEntryFormat()
    return 'WebDevIconsGetFileTypeSymbol(absolute_path) ." ". entry_path'
endfunction
```
How can I accomplish the same thing with this plugin?

I know that there is `luaeval` and tried to do the following:
```vim
return luaeval("require('nvim-web-devicons').get_icon_by_filetype( absolute_path ) .. ' entry_path'")
```
but that doesn't seem to work.

I don't want to use `vim-devicons` for that since it doesn't support colours by default and carrying two plugins doing the same thing seems ugly.

**Comment by D00mch** [comment, marked as answer]
There is an answer [on reddit](https://www.reddit.com/r/neovim/comments/o0mnxi/how_do_i_change_vimstartifys_file_icons_from/):

Web Devicons wrapper in lua:
```lua
function _G.webDevIcons(path)
  local filename = vim.fn.fnamemodify(path, ':t')
  local extension = vim.fn.fnamemodify(path, ':e')
  return require'nvim-web-devicons'.get_icon(filename, extension, { default = true })
end
```

This is the vim function for startify
```vim
function! StartifyEntryFormat() abort
  return 'v:lua.webDevIcons(absolute_path) . " " . entry_path'
endfunction
```

**Reply by Rizhiy** [reply]
Thanks, that almost works, but I'm still not getting any colour. Any idea what can fix it?

**Reply by alex-courtis** [reply]
You'll need a [patched font](https://www.nerdfonts.com/) to show the icons.

e.g. arch linux `pacman -S ttf-hack-nerd`

You'll also probably need to `set termguicolors`.

Try installing [nvim-tree.lua](https://github.com/nvim-tree/nvim-tree.lua) and see if it shows coloured icons.

**Reply by Rizhiy** [reply]
I've installed [nerd-fonts](https://github.com/ryanoasis/nerd-fonts) already.
I use [coc-explorer](https://github.com/weirongxu/coc-explorer) and icons are coloured there.
`termguicolors` are set as follows:
```
if exists('+termguicolors')
  set t_8f=^[[38;2;%lu;%lu;%lum        " set foreground color
  set t_8b=^[[48;2;%lu;%lu;%lum        " set background color
  set t_Co=256
  set termguicolors
endif
```

Do you know of any other possible issues?

Just to check: it is possible to have coloured icons in startify?

**Reply by alex-courtis** [reply]
Unfortunately I'm not familiar with startify.

Your fg/bg colours may be overriding fonts. What happens if you remove them?

You could rule out terminal issues by running the minimal nvim-tree config as per the [bug report](https://github.com/nvim-tree/nvim-tree.lua/issues/new?assignees=&labels=bug&template=bug_report.yml). Colours are functional with that setup.

**Reply by D00mch** [reply]
Actually, I also have this problem: not all the icons have color (compared to nvim-tree, bufferline, and lualine).
![image](https://user-images.githubusercontent.com/14236531/202978646-89e11e0a-5725-40e4-90e5-1109ecbea6ac.png)

For example, vim icon is colorful in lualine
![image](https://user-images.githubusercontent.com/14236531/202978810-e84fac80-883f-437d-9816-815ef45da598.png)

P.S.
Moon icon is fennel (.fnl);
I use the Neovide vim client, but everything is the same in terminal.
I don't override termguicolors.

**Reply by D00mch** [reply]
I eventually move to https://github.com/goolord/alpha-nvim for starting page and https://github.com/Shatur/neovim-session-manager for sessions. I thought I would miss startify, but I don't
![image](https://user-images.githubusercontent.com/14236531/207558692-19e86569-4459-4df5-a659-cc137eb524f6.png)

**Reply by alex-courtis** [reply]
Nice, thanks for sharing.
